nemo_aligner/data/nlp/datasets.py

```diff
@@ -323,13 +327,32 @@ def __getitem__(self, idx):
             torch.LongTensor(reject_labels), (0, max_curr_seq_len - len(reject_labels)), mode="constant", value=-100
         )
 
+        # ignore the example whose tokenized text exceeds max seq length.
+        if max_curr_seq_len > self.seq_length:
+            chosen_tokens = chosen_tokens[: self.seq_length]
+            rejected_tokens = rejected_tokens[: self.seq_length]
+            labels_chosen_tokens = torch.ones_like(chosen_tokens) * (-100)
+            labels_reject_tokens = torch.ones_like(rejected_tokens) * (-100)
+            chosen_len = self.seq_length
+            reject_len = self.seq_length
+
+        # include the ground-truth rewards of the chosen & rejected examples
+        if "chosen_reward" in payload and "rejected_reward" in payload:
+            chosen_reward = payload["chosen_reward"]
+            rejected_reward = payload["rejected_reward"]
+        else:
+            chosen_reward = self.default_chosen_reward
+            rejected_reward = self.default_rejected_reward
+
         output = {
             "chosen": chosen_tokens,
             "rejected": rejected_tokens,
             "chosen_length": chosen_len,
             "rejected_length": reject_len,
             "chosen_labels": labels_chosen_tokens,
             "rejected_labels": labels_reject_tokens,
+            "chosen_reward": chosen_reward,
+            "rejected_reward": rejected_reward,
```

Comment:
```suggestion
            "chosen_reward": payload.get("chosen_reward", self.default_chosen_reward),
            "rejected_reward": payload.get("rejected_reward", self.default_rejected_reward),
```

and then no need for the if/else above

Reply:
Done